services/github_service.py

```diff
@@ -376,6 +376,47 @@ def get_paginated_list_of_org_repository_names(self, after_cursor: str | None,
         """), variable_values={"organisation_name": self.organisation_name, "page_size": page_size,
                                "after_cursor": after_cursor})
 
+    @retries_github_rate_limit_exception_at_next_reset_once
+    def get_paginated_list_of_unlocked_unarchived_repos_and_their_first_100_outside_collaborators(
+        self,
+        after_cursor: str | None,
+        page_size: int = GITHUB_GQL_DEFAULT_PAGE_SIZE,
+    ) -> dict[str, Any]:
+        logging.info(
+            f"Getting paginated list of org unlocked unarchived repositories and their first 100 outside collaborators. Page size {page_size}, after cursor {bool(after_cursor)}"
+        )
+        if page_size > self.GITHUB_GQL_MAX_PAGE_SIZE:
+            raise ValueError(
+                f"Page size of {page_size} is too large. Max page size {self.GITHUB_GQL_MAX_PAGE_SIZE}")
+        return self.github_client_gql_api.execute(gql("""
+            query($organisation_name: String!, $page_size: Int!, $after_cursor: String) {
+                organization(login: $organisation_name) {
+                    repositories(first: $page_size, after: $after_cursor, isLocked: false, isArchived: false) {
```

Comment:
It seems you don't actually use $page_size

Reply:
It's used in line 394 setting the repo page size; I left the collaborator line hard coded to 100 to make it clearer that this part is not paginated; as the ridiculously long name suggests - it will only get the first 100 collaborators; whereas the repos are paginated to whatever the max is.